assets/js/script.js

```diff
@@ -36,26 +42,52 @@ function fn1(e) {
    var newRecipe = 'https://api.spoonacular.com/recipes/findByIngredients?ingredients='+ allIngreds + '&number=10&apiKey=39791063581a4d96a908bb19745b3f64';
  
    fetch(newRecipe)
-      .then(response => {
-         if (!response.ok){
-            throw Error("ERROR")
-         };
-         return response.json();
-   })
+  .then(response => {
+     if (!response.ok){
+        throw Error("ERROR")
+     };
+     return response.json();
+    })
    .then(data => {
-      console.log(data);
+   console.log(data);
+   if (data.length === 0) {
+     modal.style.display = "block";
+   } else {
+     modal.style.display = "none";
+   }
+   
    });
 
+   span.onclick = function() {
+      modal.style.display = "none";
+   };
+
+   window.onclick = function(event) {
+      if (event.target == modal) {
+         modal.style.display = "none";
+      };
+   };
+
    var inputs = document.querySelectorAll('#form1, #form2, #form3')
    inputs.forEach(input => {
       input.value = '';
    });
-
    e.preventDefault();
-
- 
 };
 
+//display modal pop up if no results found by length of array
+
+var modal = document.getElementById("myModal");
+var btn = document.getElementById("searchBtn");
+var span = document.getElementsByClassName("close")[0];
+var data = data
```

Comment:
This is great! My only comments are just about moving the code around so that it reads more linearly from top to bottom, that is could we please?

1. Move the .onclick functions down beneath these variables?
2. Put the if statement from lines 53-57 into a function (eg` checkNumberResults()` or something), move that function down here so it's with the other modal functions and then just call the function where the whole thing currently is now by using `checkNumberResults(data)`